Modify PowChain service

nisdas opened the pull request:
This PR is part of #1284. 

The goals of the PR are: 
- [x] Modify the PowChain Service to be up to date with the spec.
- [x] Update VRC contract.
- [x] Add Relevant tests

prestonvanloon commented on beacon-chain/powchain/service.go:
```diff
@@ -158,20 +195,61 @@ func (w *Web3Service) run(done <-chan struct{}) {
+
+func (w *Web3Service) SaveInTrie(depositData common.Hash, merkleRoot common.Hash) error {
+	if w.depositTrie.Root() != merkleRoot {
+		return errors.New("Saved root in trie is unequal to root received from log ")
```
Please format this error properly. No capitalization, no trailing whitespace

prestonvanloon commented:
Some lint issues

nisdas commented:
@prestonvanloon Lint is fixed

prestonvanloon commented on beacon-chain/powchain/service.go:
```diff
@@ -158,20 +206,108 @@ func (w *Web3Service) run(done <-chan struct{}) {
+
+}
+
+func (w *Web3Service) processChainStartLog(VRClog gethTypes.Log) {
```
```go
func (w *Web3Service) processChainStartLog(_ gethTypes.Log) {
  // TODO
}
```

prestonvanloon replied:
You will need to ignore the parameter name or the linter will complain it is unused. Please refer to the example I gave above.

prestonvanloon replied:
Nevermind, it looks like the linter doesn't care either way

terenc3t commented on beacon-chain/powchain/service.go:
```diff
@@ -158,20 +206,108 @@ func (w *Web3Service) run(done <-chan struct{}) {
+
+// processDepositLog processes the log which had been received from
+// the POW chain by trying to ascertain which participant deposited
+// in the contract
```
period.

terenc3t commented on beacon-chain/powchain/service.go:
```diff
@@ -158,20 +206,108 @@ func (w *Web3Service) run(done <-chan struct{}) {
+
+}
+
+func (w *Web3Service) processChainStartLog(VRClog gethTypes.Log) {
```
would be nice to add a comment here

terenc3t commented on beacon-chain/powchain/service.go:
```diff
@@ -158,20 +206,108 @@ func (w *Web3Service) run(done <-chan struct{}) {
+	_ = VRClog
+}
+
+func (w *Web3Service) unPackLogData(data []byte) ([]byte, []byte, error) {
```
comment here too

nisdas commented:
@terenc3t @prestonvanloon Addressed your comments , PTAL 

@terenc3t We already have a prefix for the log https://github.com/prysmaticlabs/prysm/blob/master/beacon-chain/powchain/service.go#L16
